File: chapter-01/the_dark_tower.exs
defmodule TheDarkTower do
  def capitalize_words(title) do
    title
    |> String.split()
    |> capitalize_all()
    |> join_with_whitespace()
  end

  defp capitalize_all(list) do
    Enum.map(list, &String.upcase/1)
  end

  defp join_with_whitespace(list) do
    Enum.join(list, " ")
  end
end

File: chapter-01/the_dark_tower.js
const pipe = (...fns) => (x) => fns.reduce((v, f) => f(v), x);

const split_words = (title) => title.split();

const capitalize_all = (array) => array.map((str) => str.toUpperCase());

const join_with_withespace = (array) => array.join(" ");

const capitalize_words = (title) =>
  pipe(split_words, capitalize_all, join_with_withespace)(title);
